### PR TITLE
feat(fund-from): add Fund-From tag to eligible methods/commands PE-7291

### PR DESCRIPTION
**Pull request by fedellen**

### CODE REVIEW

File: src/cli/cli.ts [hunk 1]
```diff
@@ -333,7 +328,7 @@ makeCommand<PaginationCLIOptions>({
 makeCommand<AddressAndNameCLIOptions>({
   name: 'get-primary-name',
   description: 'Get primary name',
-  options: [...addressOptions, optionMap.name],
+  options: [...[optionMap.address], optionMap.name],
```
Comment (fedellen) [inline src/cli/cli.ts:331]: looks like a mass replace error, works but unnecessary

File: src/cli/cli.ts [hunk 2]
```diff
@@ -489,124 +484,39 @@ makeCommand({
   action: redelegateStake,
 });
 
-makeCommand<BuyRecordCLIOptions>({
```
Comment (fedellen) [inline src/cli/cli.ts:492]: this PR includes a refactor that moves arns purchase actions to their own file and improves/shares their respective assertions

File: src/types/io.ts [hunk 3]
```diff
@@ -642,7 +646,10 @@ export interface AoARIOWrite extends AoARIORead {
     },
     options?: WriteOptions,
   ): Promise<AoMessageResult>;
-  requestPrimaryName(params: { name: string }): Promise<AoMessageResult>;
+  requestPrimaryName(
+    params: AoArNSPurchaseParams,
+    options?: WriteOptions,
```
Comment (fedellen) [inline src/types/io.ts:651]: adds the missing writeOptions to requestPrimaryName method

File: src/cli/cli.ts [hunk 4]
```diff
@@ -58,18 +65,15 @@ import {
 import { transfer } from './commands/transfer.js';
 import {
   addressAndVaultIdOptions,
-  addressOptions,
   antStateOptions,
+  arNSPurchaseOptions,
```
[the remark this answers is not in the document]
Comment (fedellen) [reply src/cli/cli.ts:69]: yeah it seemed odd. maybe `arnsPurchaseOptions` is a fine pattern when acronyms fall on the first word of camelcase?

**Review by atticusofsparta — DISMISSED** [review]

File: src/cli/cli.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (atticusofsparta) [reply src/cli/cli.ts:69]: Sounds good to me yup

**Pull request merged**